centreon/www/widgets/hostgroup-monitoring/src/index.php

```diff
@@ -97,161 +97,175 @@
+            );
+        }
     }
-    if ($op && isset($search) && $search != "") {
-        $query = CentreonUtils::conditionBuilder(
-            $query,
-            "name " . CentreonUtils::operandToMysqlFormat($op) . " '" . $dbb->escape($search) . "' "
-        );
+
+    if (!$centreon->user->admin) {
+        $query = CentreonUtils::conditionBuilder($query, "name IN (" . $aclObj->getHostGroupsString("NAME") . ")");
     }
-}
 
-if (!$centreon->user->admin) {
-    $query = CentreonUtils::conditionBuilder($query, "name IN (" . $aclObj->getHostGroupsString("NAME") . ")");
-}
+    $orderby = "name ASC";
+    if (isset($preferences['order_by']) && trim($preferences['order_by']) != "") {
+        $orderby = $preferences['order_by'];
+    }
 
-$orderby = "name ASC";
-if (isset($preferences['order_by']) && trim($preferences['order_by']) != "") {
-    $orderby = $preferences['order_by'];
-}
+    $query .= "ORDER BY :orderby";
+    $query .= " LIMIT " . ($page * $preferences['entries']) . "," . $preferences['entries'];
 
-$query .= "ORDER BY $orderby";
-$query .= " LIMIT " . ($page * $preferences['entries']) . "," . $preferences['entries'];
-$res = $dbb->query($query);
-$nbRows = (int) $dbb->query('SELECT FOUND_ROWS() AS REALTIME')->fetchColumn();
-$data = [];
-$detailMode = false;
-if (isset($preferences['enable_detailed_mode']) && $preferences['enable_detailed_mode']) {
-    $detailMode = true;
-}
+    $statement = $dbb->prepareQuery($query);
 
-$kernel = \App\Kernel::createForWeb();
-$resourceController = $kernel->getContainer()->get(
-    \Centreon\Application\Controller\MonitoringResourceController::class
-);
-
-$buildHostgroupUri = function (array $hostgroup, array $types, array $statuses) use ($resourceController) {
-    return $resourceController->buildListingUri(
-        [
-            'filter' => json_encode(
-                [
-                    'criterias' => [
-                        [
-                            'name' => 'host_groups',
-                            'value' => $hostgroup,
-                        ],
-                        [
-                            'name' => 'resource_types',
-                            'value' => $types,
+    $dbb->executePreparedQuery($statement, ['orderby' => [$orderby, PDO::PARAM_STR]], true);
+
+    $nbRows = (int) $dbb->executeQuery('SELECT FOUND_ROWS() AS REALTIME')->fetchColumn();
+    $data = [];
+    $detailMode = false;
+    if (isset($preferences['enable_detailed_mode']) && $preferences['enable_detailed_mode']) {
+        $detailMode = true;
+    }
+
+    $kernel = \App\Kernel::createForWeb();
+    $resourceController = $kernel->getContainer()->get(
+        \Centreon\Application\Controller\MonitoringResourceController::class
+    );
+
+    $buildHostgroupUri = function (array $hostgroup, array $types, array $statuses) use ($resourceController) {
+        return $resourceController->buildListingUri(
+            [
+                'filter' => json_encode(
```

Comment:
json_encode can return false on failure so you need to test it before use the result. You can use an exception with the flag [JSON_THROW_ON_ERROR](https://www.php.net/manual/en/json.constants.php#constant.json-throw-on-error)

Reply:
ok